compiler/luci/pass/src/FuseMulWithFullyConnectedPass.cpp

```diff
@@ -0,0 +1,183 @@
+{
+  auto node_size = fused_node->size<loco::DataType::FLOAT32>();
+  // Scalar:
+  if (multiplication->rank() == 1 ||
+      multiplication->rank() == 0 && multiplication->size<loco::DataType::FLOAT32>() == 1)
+  {
+    for (uint32_t i = 0; i < node_size; i++)
+      fused_node->at<loco::DataType::FLOAT32>(i) *= multiplication->at<loco::DataType::FLOAT32>(0);
+  }
+  // N-size:
+  else
+  {
+    for (uint32_t i = 0; i < node_size; i++)
+      fused_node->at<loco::DataType::FLOAT32>(i) *= multiplication->at<loco::DataType::FLOAT32>(i);
+  }
+}
+
+/**
+ *  Fuse Mul to FullyConnected if the multiplied value is a channel(last dimension)-wise constant
+ *
+ *  BEFORE
+ *                |
+ *      [CircleFullyConnected]
+ *                |
+ *           [CircleMul]
+ *                |
+ *
+ *  AFTER
+ *                |
+ *       [CircleFullyConnected]   [CircleMul] (dead)
+ *                |
+ *
+ */
+bool fuse_mul_with_fc(luci::CircleFullyConnected *fc)
+{
+  // Sanity check:
+  RETURN_FALSE_UNLESS(fc);
+  // Allow only FLOAT32 data type:
+  RETURN_FALSE_UNLESS(fc->dtype() == loco::DataType::FLOAT32);
+  // Allow only without activation functions as values are going to
+  // be multiplied before activation function.
+  RETURN_FALSE_UNLESS(fc->fusedActivationFunction() == luci::FusedActFunc::NONE);
+  // Check for weights being Constant:
+  auto weights = dynamic_cast<luci::CircleConst *>(fc->weights());
+  RETURN_FALSE_UNLESS(weights);
+  // Get Mul node:
+  auto fc_output = loco::succs(fc);
+  // Make sure that FullyConnected has only one child:
+  RETURN_FALSE_UNLESS(fc_output.size() == 1);
+  auto mul = dynamic_cast<luci::CircleMul *>(*fc_output.begin());
+  RETURN_FALSE_UNLESS(mul);
+  // Allow Mul node only with FLOAT32 data type:
+  RETURN_FALSE_UNLESS(mul->dtype() == loco::DataType::FLOAT32);
+  // Get multiplication Constant (here: the second input besides weights):
+  auto multiplication = mul->x() == fc ? dynamic_cast<luci::CircleConst *>(mul->y())
+                                       : dynamic_cast<luci::CircleConst *>(mul->x());
+  RETURN_FALSE_UNLESS(multiplication);
+  // Get rank of multiplication:
+  auto rank = multiplication->rank();
+  RETURN_FALSE_UNLESS(rank != 0);
+  // Check that all dimensions are ones, checks broadcast capabilites.
+  // Last dimesion of multiplication must be compatible with FC.
+  // N-D case (N>1):
+  if (multiplication->rank() > 1)
+  {
+    // Check channel-wise broadcasting:
+    for (uint32_t i = 0; i < rank - 1; i++)
+      RETURN_FALSE_UNLESS(multiplication->dim(i).value() == 1);
+  }
+  // Scalar case:
+  else if (multiplication->rank() == 1 || multiplication->rank() == 0)
+  {
+    RETURN_FALSE_UNLESS(multiplication->size<loco::DataType::FLOAT32>() != 0);
+  }
+
+  // Update weights accordingly.
+  RETURN_FALSE_UNLESS(weights->opcode() == luci::CircleOpcode::CIRCLECONST or
+                      weights->opcode() == luci::CircleOpcode::CIRCLEOUTPUTEXCLUDE)
```

Comment:
weights is always const.

Reply:
Done in scope of https://github.com/Samsung/ONE/pull/13528/commits/4c7f5a9e49824a06ae2e52997737b41019f2d259